File: Program43.cpp
#include<iostream>
using namespace std;

class Factors
{
    private:
       int iNo;
    
    public:
        void Accept()
        {
            cout<<"Enter number"<<endl;
            cin>>this->iNo;
        }
        void Display()
        {
            cout<<"Entered value is: "<<this->iNo<<endl;
        }
        bool CheckPrime()
        {
            int iCnt = 0;

            for(iCnt = 2; iCnt <= (iNo/2); iCnt++)
            {
                if((iNo % iCnt) == 0)
                {
                    return false;
                }
            }
            return true;
        }
};

int main()
{
    bool bRet;
    Factors fobj;

    fobj.Accept();
    fobj.Display();

    bRet = fobj.CheckPrime();

    if(bRet == true)
    {
        cout<<"Prime number"<<endl;
    }
    else
    {
        cout<<"Not prime numer"<<endl;
    }
    
    return 0;
}

File: Program43.java
import java.lang.*;
import java.util.*;

class Factors
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number");
        this.iNo = sobj.nextInt();
    }
    public void Display()
    {
        System.out.println("Entered value is: "+this.iNo);
    }
    public boolean CheckPrime()
    {
        int iCnt = 0;

        for(iCnt = 2; iCnt <= (iNo/2); iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                return false;
            }
        }
        return true;
    } 
}

class Program43
{
    public static void main(String arg[])
    {
        boolean bRet;
        Factors fobj = new Factors();

        fobj.Accept();
        fobj.Display();
        
        bRet = fobj.CheckPrime();

        if(bRet == true)
        {
            System.out.println("Prime number");
        }
        else
        {
            System.out.println("Not prime number");
        }

    }
}